stream.go

```diff
@@ -25,6 +27,11 @@ type Stream[T any] interface {
 	// the current value.
 	WaitNext() T
 
+	// WaitNextCtx waits for Changes to be closed or the passed context to be canceled.
+	// When Changes is closed first, it advances the stream and returns the current value.
+	// When the context is canceled first, the error recieved from the context is returned.
```

Comment:
I find the comment a bit complex. What about

```
WaitNextCtx does the same as WaitNext but returns earlier with an error if the given context is cancelled first.
```

Reply:
Sounds sensible, I will change this :)